frontend/tests/badges/cppCourseGraduateTest.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Simple test for legacy user Badge
+ *
+ * @author RuizYugen
+ */
+class CppCourseGraduate extends \OmegaUp\Test\BadgesTestCase {
+    protected $courseData;
+    protected $students;
+    protected $submissionSource;
+    protected $problems;
+
+    public function setUp(): void {
+        parent::setUp();
+        $courseAlias = 'introduccion_a_cpp';
+
+        //create course
+        $this->courseData = \OmegaUp\Test\Factories\Course::createCourseWithAssignments(
+            /*$nAssignments=*/ 2,
+            $courseAlias
+        );
+        $studentsInCourse = 2;
+
+        // Prepare assignment. Create four problems: The first three accept
+        // submissions and the last one does not.
+        $adminLogin = self::login($this->courseData['admin']);
+        $this->problems = [];
+        for ($i = 0; $i < 3; $i++) {
+            $this->problems[] = \OmegaUp\Test\Factories\Problem::createProblem();
+        }
+        $this->problems[] = \OmegaUp\Test\Factories\Problem::createProblem(
+            new \OmegaUp\Test\Factories\ProblemParams([
+                'languages' => '',
+            ])
+        );
+
+        foreach (array_slice($this->problems, 0, 2) as $problemData) {
+            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $this->courseData['course_alias'],
+                'assignment_alias' => $this->courseData['assignment_aliases'][0],
+                'problem_alias' => $problemData['request']['problem_alias'],
+            ]));
+        }
+        foreach (array_slice($this->problems, 2, 2) as $problemData) {
+            \OmegaUp\Controllers\Course::apiAddProblem(new \OmegaUp\Request([
+                'auth_token' => $adminLogin->auth_token,
+                'course_alias' => $this->courseData['course_alias'],
+                'assignment_alias' => $this->courseData['assignment_aliases'][1],
+                'problem_alias' => $problemData['request']['problem_alias'],
+            ]));
+        }
+
+        // Add students to course
+        $this->students = [];
+        for ($i = 0; $i < $studentsInCourse; $i++) {
+            $this->students[] = \OmegaUp\Test\Factories\Course::addStudentToCourse(
+                $this->courseData
+            );
+        }
+
+        $this->submissionSource = "#include <stdio.h>\nint main() { printf(\"3\"); return 0; }";
+    }
+
+    public function testCourseCppUserEarnBadge() {
+        {
+            // This user will recive the badge because he will resolve 4 differents problems of the course
```

Comment:
nit:

```suggestion
            // This user will receive the badge because they will resolve 4 differents problems
            // of the course
```

hay que evitar:

* Errores de ortografía
* Asignarle un género al usuario (siempre hay que usar "they" en vez de "he").
* Comentarios que excedan ~80 caracteres.